lib/linalg/sparse_vector.cc

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include "clink/linalg/sparse_vector.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+using namespace clink;
+using namespace std;
+
+SparseVector::SparseVector(const int n, const int * indices, const int indicesLen, const double * values, const int valuesLen) {
+    this->n = n;
+    this->indicesLen = indicesLen;
+    this->indices = new int[this->indicesLen];
+    for (int i = 0; i < this->indicesLen; i++) {
+      this->indices[i] = indices[i];
+    }
+    this->valuesLen = valuesLen;
+    this->values = new double[this->valuesLen];
+    for (int i = 0; i < this->valuesLen; i++) {
+      this->values[i] = values[i];
+    }
+}
+
+SparseVector::~SparseVector(void) {
+  delete [] this->indices;
```

Comment:
Would it be simpler to use `std::unique_ptr` as the type of `indices` instead of explicitly deleting it here?

Same for `values`.

And could you add empty lines as appropriate? Maybe follow TFRT's code style for?

Does `~SparseVector` needs to be `public`? Same for other methods/variables.

Reply:
OK. I'll try it.

Reply:
Changing the types to `std::unique_ptr` seems not feasible. JNA can only identify certain kinds of C++ data types and `std::unique_ptr` is not included.

I didn't see deconstructors in TFRT add much empty lines, so I think my current implementation follows TFRT's code style. Still, there are empty lines and member order formats in structs' definitions in header files of TFRT, so I make adjustments accordingly in those places.

`~SparseVector` is invoked externally when a `SparseVector` is returned from `OneHotEncoderModel_transform()` method, so it needs to be public.